File: android/app/src/main/java/com/reservationapp/ui/admin/components/WeeklySummaryView.kt
package com.reservationapp.ui.admin.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.horizontalScroll
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.reservationapp.core.ui.theme.BackgroundInactiveTab
import com.reservationapp.core.ui.theme.BackgroundPaper
import com.reservationapp.core.ui.theme.BorderDefault
import com.reservationapp.core.ui.theme.Spacing
import com.reservationapp.core.ui.theme.TextPrimary
import com.reservationapp.core.ui.theme.TextSecondary
import com.reservationapp.core.ui.theme.TextTertiary
import com.reservationapp.domain.model.DailySummary

@Composable
fun WeeklySummaryView(
    dailyData: List<DailySummary>,
    isLoading: Boolean,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundPaper
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(Spacing.lg)
        ) {
            Text(
                text = "Haftalık Rezervasyon Özeti",
                style = MaterialTheme.typography.titleMedium.copy(
                    fontWeight = FontWeight.SemiBold
                ),
                color = TextPrimary
            )

            if (isLoading) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(Spacing.xl),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            } else {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .horizontalScroll(rememberScrollState()),
                    horizontalArrangement = Arrangement.spacedBy(Spacing.md)
                ) {
                    dailyData.forEach { day ->
                        DailySummaryCard(day = day)
                    }
                }
            }
        }
    }
}

@Composable
private fun DailySummaryCard(
    day: DailySummary
) {
    Card(
        modifier = Modifier.width(100.dp),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundInactiveTab
        ),
        border = androidx.compose.foundation.BorderStroke(1.dp, BorderDefault)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.md),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            Text(
                text = day.dayAbbreviation,
                style = MaterialTheme.typography.bodyMedium.copy(
                    fontWeight = FontWeight.SemiBold
                ),
                color = TextSecondary
            )
            Text(
                text = "${day.reservationCount}",
                style = MaterialTheme.typography.titleLarge.copy(
                    fontWeight = FontWeight.Bold
                ),
                color = TextPrimary
            )
            Text(
                text = "Rezervasyon",
                style = MaterialTheme.typography.labelSmall,
                color = TextTertiary
            )
        }
    }
}

File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Dashboard/WeeklySummaryView.swift
//
//  WeeklySummaryView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct WeeklySummaryView: View {
    let dailyData: [DailySummary]
    let isLoading: Bool
    
    var body: some View {
        VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.md) {
            Text("Haftalık Rezervasyon Özeti")
                .font(AppTypography.h6())
                .fontWeight(.semibold)
                .foregroundColor(AppColors.textPrimary)
            
            if isLoading {
                ProgressView()
                    .frame(maxWidth: .infinity)
                    .padding(ThemeManager.shared.spacing.xl)
            } else {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: ThemeManager.shared.spacing.md) {
                        ForEach(dailyData) { day in
                            DailySummaryCard(day: day)
                        }
                    }
                }
            }
        }
        .padding(ThemeManager.shared.spacing.lg)
        .background(AppColors.backgroundPaper)
        .cornerRadius(ThemeManager.shared.borderRadius.card.value)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
}

struct DailySummaryCard: View {
    let day: DailySummary
    
    var body: some View {
        VStack(spacing: ThemeManager.shared.spacing.xs) {
            Text(day.dayAbbreviation)
                .font(AppTypography.body2())
                .fontWeight(.semibold)
                .foregroundColor(AppColors.textSecondary)
            
            Text("\(day.reservationCount)")
                .font(AppTypography.h5())
                .fontWeight(.bold)
                .foregroundColor(AppColors.textPrimary)
            
            Text("Rezervasyon")
                .font(AppTypography.caption())
                .foregroundColor(AppColors.textTertiary)
        }
        .frame(width: 100)
        .padding(ThemeManager.shared.spacing.md)
        .background(AppColors.backgroundInactiveTab)
        .cornerRadius(ThemeManager.shared.borderRadius.button.value)
        .overlay(
            RoundedRectangle(cornerRadius: ThemeManager.shared.borderRadius.button.value)
                .stroke(AppColors.borderDefault, lineWidth: 1)
        )
    }
}



